### ISSUE TITLE
Newsletter subscriptions status not isolated between multi stores

### ISSUE THREAD

**Issue by mikelevy300** [magento/magento2#10014]
When a customer with the same email address has an account on different multi stores in the same Magento installation, changes to the newsletter subscription in one account affect the other.

### Preconditions
1. Magento 2.1.5
2. PHP 7.0.16
3. MySQL 5.7
4. Apache 2.4
5. At least two stores setup with nothing shared
6. A customer exists in both stores with the same email address 

### Steps to reproduce
1. Login with customer 1 in store A, go to My Account -> Newsletter Subscriptions, and make sure he's subscribed to the general newsletter
2. Login with customer 2 in store B, go to My Account -> Newsletter Subscriptions, and make sure he's subscribed to the general newsletter
3. Unsubscribe customer 1 from the newsletter
4. Refresh the page for customer 2

### Expected result
1. Customer 1 is unsubscribed
2. Customer 2 is subscribed

### Actual Result
1. Customer 1 is unsubscribed
2. Customer 2 is unsubscribed

### Extra info
When looking in the newsletter_subscriber table, instead of there being one record for each customer per store, there's one global record.  When customer 1 in store A updates, the store_id in newsletter subscriber is updated to store A's ID.  When customer 2 in store B updates, the store_id in newsletter_subscriber is updated to store B's ID.  

Tracing the code out, I think the problem is in Magento_Newsletter\Model\ResourceModel\Subscriber.php, specifically the loadByCustomerData function.  
Here's the function:

```php
public function loadByCustomerData(\Magento\Customer\Api\Data\CustomerInterface $customer){
	$select = $this->connection->select()->from($this->getMainTable())->where('customer_id=:customer_id');

	$result = $this->connection->fetchRow($select, ['customer_id' => $customer->getId()]);

	if ($result) {
		return $result;
	}

	$select = $this->connection->select()->from($this->getMainTable())->where('subscriber_email=:subscriber_email');

	$result = $this->connection->fetchRow($select, ['subscriber_email' => $customer->getEmail()]);

	if ($result) {
		return $result;
	}

	return [];
}
```

It basically uses the customerId as the lookup key but then falls back to the customerEmail if nothing is found.  This creates the unintended consequence where updates in one store show in the other.  A suggested fix would be to either only use the customerId or add an additional field to the select that matches the store_id of the customer, something like

```
select * from newsletter_subscriber where subscriber_email = $customer->getEmail() and store_id = $customer->getStoreId()
```

**Comment by AlexWorking** [comment]
Hello @mikelevy300. Based on Your issue an internal ticket was created - MAGETWO-70081. You'll be informed right after Your issue is resolved. Thank You for applying.

**Comment by magento-engcom-team** [comment]
@mikelevy300, thank you for your report.
We've created internal ticket(s) MAGETWO-70081 to track progress on the issue.

**Comment by sbaixauli** [comment]
I am working on it at #mm17es

**Comment by okorshenko** [comment]
The issue has been fixed and delivered to 2.2-develop branch. Will be available with 2.2.2 release

**Comment by TomashKhamlai** [comment]
Not "stores" but "websites"

**Comment by magento-engcom-team** [comment]
Hi @mikelevy300. Thank you for your report. 
The issue has been fixed in magento-engcom/magento2ce#1289 by @magento-engcom-team in 2.3-develop branch
Related commit(s):
 - b41594122f532f6a89165a70906a5b257562c0df
 - 7369261d82e0e5ad926eaf4ccb2a8def79c4fcbb
 - dce1c66301cdddfee2dc2887f584c333d0b438b2
 - 95ed67ee4adbbd7f3f0107a52da4b9238cf44803
 - 05fcdd1a52c674ea09a34960cea71e73488fbc99
 - 8b1a2972b3f0ddd4713d00055b8d60f737a72ce4
 - 90addb731c3b253d3da4dc564ddf94ac3525bd0f
 - d5c9eef075cd5f4d9781ec7f3151a14e087d061d
 - 6ec62717c16ed96d45e670d7f867b724f0449aaa
 - f0af87edad79b295b4be87f6f92502bacfbc2764
 - 2497062749bf41c80aaebb9b6cd21eb48656f58c
 - 1f7347f47cca48238f9deabd48a0a280bc0c309f
 - 0d151c41539ba7c4d20ede20d614bb4be257903c
 - 07982fdea19074788654c6e825874cb8188ab627
 - 617da6ca08607890779ce71530a2871438c88351
 - 39ed21f83071ebffd71963e9097ecb59fd906702
 - 9aa036dec42211867f68f714a746b71395c43bb2
 - fda37ef0af75e6b28d4a0da1781b19c7a7894a97
 - 47b109180b6b5e499ba8d82e6c67888fca21ffe8
 - ef6badace5c6b046047a6e5f617020f9d4410a82
 - 809ec050986fa3ddfeec4a715ce5ace1e41a4427

The fix will be available with the upcoming patch release.

**Comment by dianacle** [comment]
Hello, is this a regression or a breaking change in version > 2.2?

I tested the newsletter subscription in many version of magento ce.
It seems to me that:
- in magento <= 2.1 (including magento 1.x) > newsletter subscription are unique per email (guest or customer doesn't matter, subscribing the same email for different store_id do overwrite  previous record)
- in magento version > 2.2 > newsletter subscription are unique per email/website (can subscribe the same email for different store_id of different websites, but not for store_id of same website)
